docs/PFCWD-test-plan.md

```diff
@@ -0,0 +1,189 @@
+
+* All existing packets in the output queue are discarded.
+* All subsequent packets destined to the output queue are discarded.
+* All subsequent packets received by the corresponding priority group of this queue are discarded including the pause frames received. As a result, the switch should not generate any pause frame to its neighbor due to congestion of this output queue.
+  
+When forward action is selected, following actions need to be implemented:
+
+* The queue no longer honors the PFC frames received. All packets destined to the queue are forwarded along with those packets that were in the queue.
+
+### PFC Storm Restoration
+
+The watchdog should continue count the PFC frames received on the queue. If there is no PFC frame received over T1 period. Then, re-enable the PFC on the queue and stop dropping packets if the previous mitigation was drop. T1 is port level parameter. T1 should be on the scale of hundred milliseconds.
+
+### PFC Polling Interval 
+
+PFC Polling interval is the the frequency at which queues should be checked for congestion or recovery detection. By default, polling interval is calculated internally or it considers the value configured through CLI.
+
+### 1.1.1 Scope
+
+The test cases depicted in this document aim to do functional testing of PFC watchdog behavior of SONiC DUT (Device Under Test).
+
+### 1.1.2 Testbed
+#### 1.1.2.1 Topology 1
+```
++-----------------+           +--------------+           +-----------------+       
+| Keysight Port 1 |------ et1 |   SONiC DUT  | et2 ------| Keysight Port 2 | 
++-----------------+           +--------------+           +-----------------+ 
+                                   et3
+                                    |
+                                    |
+                                    |
+                            +-----------------+
+                            | Keysight Port 3 |
+                            +-----------------+
+```
+*Fig. 1: Topology 1*
+
+## 1.2 Setup configuration
+
+### 1.2.1 DUT Configuration
+•PFC watchdog and PFC must be enabled in the DUT by default.
+
+
+### 1.2.2 Keysight configuration
+•All Keysight ports should have the same bandwidth capacity.
+
+•	Test specific configurations are mentioned in respective test cases.
+
+## 1.3 Test Cases
+
+### 1.3.1 Test Case #1 - PFCWD two senders two receivers
+
+#### 1.3.1.1 Test Objective
+
+This test aims to verify how PFC watchdog can handle PFC storms in a topology with two senders and two receivers.
+
+#### 1.3.1.2 Test Topology
+
+Refer to Topology 1.1.2.1 for the test topology.
+
+#### 1.3.1.3 Test Configuration
+
+- On SONiC DUT configure the following:
+  1. Enable watchdog with default storm detection time (400ms) and restoration time (2sec). Please note that different switches have different detection times and restoration times, e.g., 200ms and 400ms.  
```

Comment:
Do not specify any concrete values here! Different platforms use different detection time, restoration time and polling interval

Reply:
Done